<a href="https://colab.research.google.com/github/ehardi19/Final-Task-Machine-Learning/blob/master/Tugas_1301174099_1301174223.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Tugas Besar Pembelajaran Mesin (CSH3L3)**
Kelas IFIK-40-01 Semester Ganjil Tahun Ajaran 2019/2020



**Anggota** 


1.   Eko Hardiyanto (1301174099)
2.   Ganesha Danu Enastika (1301174223)



In [0]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

## **Dataset**

In [2]:
!wget -O kc_house_data.csv https://raw.githubusercontent.com/Jonasyao/Machine-Learning-Specialization-University-of-Washington-/master/Regression/Assignment_four/kc_house_data.csv


--2019-12-07 06:02:32--  https://raw.githubusercontent.com/Jonasyao/Machine-Learning-Specialization-University-of-Washington-/master/Regression/Assignment_four/kc_house_data.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 368640 (360K) [text/plain]
Saving to: ‘kc_house_data.csv’

kc_house_data.csv   100%[===================>] 360.00K  --.-KB/s    in 0.02s   

2019-12-07 06:02:32 (22.3 MB/s) - ‘kc_house_data.csv’ saved [368640/368640]



Baca dataset dan lihat 5 data index teratas

In [3]:
df = pd.read_csv('kc_house_data.csv')
df.head()

id             date     price  ...     long  sqft_living15  sqft_lot15
0  7129300520  20141013T000000  221900.0  ... -122.257         1340.0      5650.0
1  6414100192  20141209T000000  538000.0  ... -122.319         1690.0      7639.0
2  5631500400  20150225T000000  180000.0  ... -122.233         2720.0      8062.0
3  2487200875  20141209T000000  604000.0  ... -122.393         1360.0      5000.0
4  1954400510  20150218T000000  510000.0  ... -122.045         1800.0      7503.0

[5 rows x 21 columns]

Cek dimesi dari dataset

In [4]:
df.shape

(3165, 21)

Cek informasi tipe data dari setiap kolomnya

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3165 entries, 0 to 3164
Data columns (total 21 columns):
id               3165 non-null int64
date             3165 non-null object
price            3165 non-null float64
bedrooms         3165 non-null int64
bathrooms        3165 non-null float64
sqft_living      3165 non-null int64
sqft_lot         3165 non-null int64
floors           3165 non-null float64
waterfront       3165 non-null int64
view             3165 non-null int64
condition        3165 non-null int64
grade            3165 non-null int64
sqft_above       3165 non-null int64
sqft_basement    3165 non-null int64
yr_built         3165 non-null int64
yr_renovated     3165 non-null int64
zipcode          3165 non-null int64
lat              3165 non-null float64
long             3165 non-null float64
sqft_living15    3164 non-null float64
sqft_lot15       3164 non-null float64
dtypes: float64(7), int64(13), object(1)
memory usage: 519.4+ KB


Berdasarkan informasi di atas, pada kolom `sqft_living15` dan `sqft_lot15` terdapat masing-masing 1 buah index dengan `null` value

In [6]:
df.iloc[df[df['sqft_living15'].isnull()].index]

id             date     price  ...   long  sqft_living15  sqft_lot15
3164  6021503830  20140620T000000  480000.0  ... -122.3            NaN         NaN

[1 rows x 21 columns]

In [7]:
df.iloc[df[df['sqft_lot15'].isnull()].index]

id             date     price  ...   long  sqft_living15  sqft_lot15
3164  6021503830  20140620T000000  480000.0  ... -122.3            NaN         NaN

[1 rows x 21 columns]

Berdasarkan analisis diatas data pada index 3164 memiliki `sqft_living15` dan `sqft_lot15` bernilai `NaN`

<p>Copyright &copy; 2019 <a href=https://www.linkedin.com/in/andityaarifianto/>ADF</a> </p>